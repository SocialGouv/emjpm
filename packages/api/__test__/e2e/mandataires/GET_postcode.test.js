//

const request = require("supertest");

const { knex } = global;
jest.setMock("@emjpm/api/db/knex", knex);

const server = require("@emjpm/api/app");

const { getTokenByUserType } = require("../utils");

//

beforeAll(async () => {
  await knex.migrate.latest();
  await knex.seed.run();
});

test("should get mandataires for postcode 62000", async () => {
  const token = await getTokenByUserType("mandataire");
  const response = await request(server)
    .get("/api/v1/mandataires/postcode/62000")
    .set("Authorization", "Bearer " + token);

  expect(response.body).toMatchSnapshot();
  expect(response.status).toBe(200);
});
