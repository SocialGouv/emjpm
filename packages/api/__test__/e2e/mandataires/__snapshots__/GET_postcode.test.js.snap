// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get mandataires for postcode 62000 1`] = `
Object {
  "code_postal": "62000",
  "id": 1,
  "latitude": 50.5333,
  "longitude": 2.6333,
}
`;
